### PR TITLE
Switch to `std:future::Future` and `async/await`.

### PR DESCRIPTION
**Pull request by reitermarkus**
The client part has basically remained the same, the server part however has been completely replaced with an extension trait on Hyper's `Server`.

Currently depends on Rust nightly, Tokio master and Hyper master.

### CODE REVIEW

**Comment by softprops** [comment]
This looks super dope @reitermarkus. I just need a bit of time to catch up on this works and I'll merge

File: README.md [hunk 1]
```diff
@@ -1,121 +1,105 @@
...
-fn hello(_: Request<Body>) -> impl futures::Future<Item = Response<Body>, Error = io::Error> + Send {
-    futures::future::ok(
-        Response::builder()
-            .header(header::CONTENT_TYPE, "text/plain")
-            .header(header::CONTENT_LENGTH, PHRASE.len() as u64)
-            .body(PHRASE.into())
-            .expect("failed to create response")
-    )
-}
+#[hyper::rt::main]
```
Comment (softprops) [inline README.md]: Where is this hyper::rt::main attribute defined?

Comment (reitermarkus) [reply README.md]: It's a re-export of `tokio::main`, here: https://github.com/hyperium/hyper/blob/f0478c62677a953aae84aa8d976bec1d28ef21b1/src/rt.rs#L14

Comment (reitermarkus) [reply README.md]: Seem's like it's been removed.

**Comment by softprops** [comment]
Is there an eta on when this could be releaseable on stable rust?

**Comment by reitermarkus** [comment]
This is stabilized in Rust 1.39.

**Comment by softprops** [comment]
Looks like hyper had a release https://seanmonstar.com/post/187493499882/hyper-alpha-supports-asyncawait

**Comment by ebkalderon** [comment]
@reitermarkus Async/await is stable and out in the wild! I was about to open a pull request against this repo, but I noticed this one already existed. Is it still being actively worked on? I would really like to see this crate brought to feature parity with upstream `hyper`.

**Comment by danieleades** [comment]
i hadn't seen this branch before opening https://github.com/softprops/hyperlocal/pull/29

what's the status of this?

**Comment by reitermarkus** [comment]
> what's the status of this?

Depends on what @ebkalderon wants to do here. Either release an alpha with this or wait for `hyper` 0.13 to be stable.

**Comment by ebkalderon** [comment]
@reitermarkus I didn't realize I was blocking a decision, haha. I believe an alpha release wouldn't hurt for the sake of letting intrepid users iterate faster, as long as transitioning to stable `hyper` is easy to achieve and anticipated breakage is relatively minimal. It is labeled an alpha release, after all.

With that said, I believe that final judgment should be reserved by the PR author and project owner(s) in this case, and I would equally respect the decision to wait, especially if the next release is right around the corner.

**Comment by reitermarkus** [comment]
@ebkalderon, I actually meant to tag @softprops here, sorry. 😉

**Comment by reitermarkus** [comment]
I do agree though that an alpha wouldn't hurt to get the ball rolling and that any future changes will be relatively minimal at this stage compared to all of the changes in this PR.

**Comment by softprops** [comment]
It looks the the ecosystem is starting to fall into place. Tokio recently released an [official non alpha 0.2.x release line](https://github.com/tokio-rs/tokio/blob/master/tokio/CHANGELOG.md#022-november-29-2019) and it seems the futures crate released an [official non-alpha 0.3.x release line](https://github.com/rust-lang/futures-rs/blob/master/CHANGELOG.md#031---2019-11-7) 

This pull seems to be the most complete changeset so I'll merge this. It's a holiday weekend for me so I may not get to it this weekend but I think it would be a good idea to bring the cargo deps in this project up to the latest stable versions now available in the ecosystem.

**Pull request merged**